### ISSUE TITLE
Bug in the python server generated code

### ISSUE THREAD

**Issue by zaibon** [Jumpscale/go-raml#365]
Some variable name get the array notation of raml type`[]` appended to their name. This is not a valid notation.

e.g: 
```python
Service[]_schema =  JSON.load(open('./schema/Service[]_schema.json'))
Service[]_schema_resolver = jsonschema.RefResolver('file://' + dir_path + '/schema/', Service[]_schema)
Service[]_schema_validator = Draft4Validator(Service[]_schema, resolver=Service[]_schema_resolver)

TaskCreate_schema =  JSON.load(open('./schema/TaskCreate_schema.json'))
TaskCreate_schema_resolver = jsonschema.RefResolver('file://' + dir_path + '/schema/', TaskCreate_schema)
TaskCreate_schema_validator = Draft4Validator(TaskCreate_schema, resolver=TaskCreate_schema_resolver)

Task[]_schema =  JSON.load(open('./schema/Task[]_schema.json'))
Task[]_schema_resolver = jsonschema.RefResolver('file://' + dir_path + '/schema/', Task[]_schema)
Task[]_schema_validator = Draft4Validator(Task[]_schema, resolver=Task[]_schema_resolver)
```

Here is raml file use to generate this code:

```raml
#%RAML 1.0
title: "ZeroRobot"
description: API of the ZeroRobot, used for management and communication between ZeroRobots
version: 0.0.1
mediaType: application/json
documentation:
  - title: Extended documentation
    content: https://github.com/Jumpscale/zerorobot/tree/first_draft/docs

types:
  Blueprint:
    description: |
      A blueprint is a description of which service you want to create
      and what action you want to add to a task list.
    properties:
      content: object
    example:
      content:
        node__node1:
          ip: '192.168.1.10'
          hostname: 'node1'
        actions:
          template: node
          service: node1
          action: start

  TemplateRepository:
    properties:
      url:
        type: string
        description: |
          URL of the git repository to clone.
          Need to be under one of these format
          - git@github.com:account/repository.git
          - https://github.com/account/repository.git
      branch:
        type: string
        description: branch of the repository to clone. If not specified, default to master
        default: master
        required: false
    example:
      url: https://github.com/account/repository.git
      branch: master

  Template:
    properties:
      version:
        type: string
      name:
        type: string
    example:
      version: 1.0.1
      name: node.zero-os


  ServiceState:
    properties:
      category:
        type: string
      tag:
        type: string
      state:
        enum: [ ok, error ]
    example:
      category: network
      tag: local-tcp-443
      state: ok

  ServiceCreate:
    properties:
      template:
        type: string
        description: This is the full qualified name of a template using the URL from it's repository
      version:
        type: string
      name:
        type: string
      data:
        type: object
        description: Instance data used to create the service
    example:
      template: github.com/account/repository/master/node.zero-os
      version: 1.0.1
      name: node1
      data:
        ip_addr: 192.168.1.10
        port: 6379

  Service:
    properties:
      template:
        type: string
        description: This is the full qualified name of a template using the URL from it's repository
      version:
        type: string
      name:
        type: string
      state:
        type: ServiceState[]
      actions:
        type: Action[]
    example:
      template: github.com/account/repository/master/node.zero-os
      version: 1.0.1
      name: node1
      actions:
        - name: start
        - name: stop
      state:
      - category: network
        tag: local-tcp-6379
        state: ok
      - category: network
        tag: local-http-80
        state: ok
      - category: data
        tag: db
        state: ok


  Action:
    properties:
      name:
        type: string
      arguments:
        type: string[]
        required: false
    example:
      name: start

  TaskCreate:
    properties:
      template_name:
        type: string
      service_name:
        type: string
      service_guid:
        type: string
      action_name:
        type: string
      state:
        enum: [ new, error ]
        description: |
          a task that is waiting to be executed is in state new.
          a task that failed to execute and is being retried is in state error.
    example:
      template_name: github.com/account/repository/master/node.zero-os
      service_name: node1
      service_guid: 10b0c69f-0f72-4e41-b146-4e10997024f8
      action_name: start
      state: new

  Task:
    type: TaskCreate
    description: Type return after a task is added to a task list
    properties:
      position:
        type: integer
        description: Position of the task in the list
      guid:
        type: string
        description: unique ID of the task
    example:
      template_name: github.com/account/repository/master/node.zero-os
      service_name: node1
      service_guid: 10b0c69f-0f72-4e41-b146-4e10997024f8
      action_name: start
      state: new
      position: 2
      guid: 07f38c46-056b-420d-8964-6efd3f493fe5

  Error:
    properties:
      code:
        type: integer
        description: error code
      message:
        type: string
        description: Human friendly error message
      stack_trace:
        type: string
        description: optional stack_trace attached with the error.
        required: false

/blueprints:
  description: |
    A blueprint is a description of which service you want to create
    and what action you want to add to a task list.
  post:
    displayName: ExecuteBlueprint
    description: Execute a blueprint on the ZeroRobot
    body:
      type: Blueprint
    responses:
      204:
        description: |
          Blueprint executed successfully
          Which mean all the service have been created and all actions added to the task list

/templates:
  description: |
    Templates are the source of the services. A service is just an instance of a template.
    To be able to create services, you need to first make some template available to the ZeroRobot by cloning
    a git repository that contains template.
    Once the templates are available, you can send blueprint with the description of which service to create.
  get:
    displayName: ListTemplates
    description: List all the templates available to the ZeroRobot
    responses:
      200:
        body:
          type: Template[]
  post:
    displayName: AddTemplateRepo
    description: Clone a template repository and make the templates available to the ZeroRobot
    body:
      type: TemplateRepository
    responses:
      201:
        description: Template successfully added
        body:
          description: List of all new templates
          type: Template[]
/services:
  description: |
    Services are the core of the ZeroRobot. It is the services that hold state, data and execute actions.
  get:
    displayName: listServices
    description: List all the services known by the ZeroRobot.
    body:
      type: Service[]
  post:
    displayName: createService
    description: create a new service
    body:
      type: ServiceCreate
    responses:
      201:
        description: Service created successfully
        body:
          type: Service
        headers:
          Location:
            description: URL to the created service resource
      409:
        description: Conflict, a service with this name already exists in this ZeroRobot.
  /{service_guid}:
    get:
      displayName: GetService
      description: Get the detail of a service
      responses:
        200:
          body:
            type: Service
        204:
          description: Service with this guid not found
          body:
            type: Error
    /actions:
      get:
        displayName: ListActions
        description: List all the possible action a service can do.
        responses:
          200:
            body:
              type: Action[]
    /task_list:
      description: A task list is all the actions that are scheduled for a service
      get:
        displayName: getTaskList
        description: Return all the action in the task list
        body:
          type: Task[]
      post:
        displayName: AddTaskToList
        description: Add a task to the task list
        body:
          type: TaskCreate
        responses:
          201:
            description: Task added to the task list successfully
            body:
              type: Task
            headers:
              Location:
                description: description URL to the created task resource
      /{task_guid}:
        get:
          displayName: GetTask
          description: Retrieve the detail of a task
          responses:
            200:
              body:
                type: Task
            404:
              description: Not task found in the task list with this guid
              body:
                type: Error
```

**Comment by iwanbk** [comment]
Looks like i found the issue for below generated code

```python
Service[]_schema =  JSON.load(open('./schema/Service[]_schema.json'))
```

It is because the `Service[]` definition is in request body.
If we create new type like below
```yaml
ServiceArr
  type: Service[]
```
then go-raml will generate the code properly.

It seems that go-raml doesn't have proper unit test for this, because IIRC we didn't have this issue before.

**Comment by iwanbk** [comment]
A WIP can be found on `fix-365` branch.

> It seems that go-raml doesn't have proper unit test for this, because IIRC we didn't have this issue before.

after further check, it seems to not a regression, it is indeed new issue.
We didn't get it before because we used flask-wtform

**Comment by iwanbk** [comment]
closed by #376
